platform/darwin/test/MGLStyleValueTests.swift

```diff
@@ -1,8 +1,53 @@
 import XCTest
 import Mapbox
 
-
+#if os(iOS) || os(watchOS) || os(tvOS)
+typealias ColorType = UIColor
+#elseif os(macOS)
+typealias ColorType = NSColor
+#endif
+    
 extension MGLStyleValueTests {
+    func assertColorsEqualWithAccuracy(_ actual: ColorType, _ expected: ColorType, accuracy: Float = 0.1) {
+        var actualComponents : [CGFloat] = [0, 0, 0, 0]
+        var expectedComponents : [CGFloat] = [0, 0, 0, 0]
+        actual.getRed(&(actualComponents[0]), green: &(actualComponents[1]), blue: &(actualComponents[2]), alpha: &(actualComponents[3]))
+        expected.getRed(&(expectedComponents[0]), green: &(expectedComponents[1]), blue: &(expectedComponents[2]), alpha: &(expectedComponents[3]))
+        for (ac, ec) in zip(actualComponents, expectedComponents) {
+            XCTAssertEqualWithAccuracy(Float(ac), Float(ec), accuracy: accuracy)
+        }
+    }
+    
+    func assertColorValuesEqual(_ actual: MGLStyleValue<ColorType>, _ expected: MGLStyleValue<ColorType>) {
+        if let actualConstant = actual as? MGLStyleConstantValue<ColorType> {
+            XCTAssertTrue(expected is MGLStyleConstantValue<ColorType>)
+            assertColorsEqualWithAccuracy(actualConstant.rawValue, (expected as! MGLStyleConstantValue<ColorType>).rawValue)
+            return
+        } else if let actualFunction = actual as? MGLStyleFunction<ColorType> {
+            XCTAssertTrue(expected is MGLStyleFunction<ColorType>)
+            let expectedFunction = expected as! MGLStyleFunction<ColorType>
+            XCTAssertEqual(actualFunction.interpolationBase, expectedFunction.interpolationBase)
+            XCTAssertEqual(actualFunction.interpolationMode, expectedFunction.interpolationMode)
+            // TODO: assert default values are equal
+            if actualFunction.stops == nil {
+                XCTAssertNil(expectedFunction.stops)
+                return
```

Comment:
Use a `guard … else` statement for an early return. Better yet, use a `guard let … else` statement for an early return conditioned on the existence of a variable you want to make use of below.